pgmpy/tests/test_factors/test_Factor.py

```diff
@@ -54,10 +54,20 @@ def setUp(self):
         card3 = [3, 3, 3, 2, 2, 2, 2, 2, 2]
         self.phi3 = Factor(['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
                            card3, np.arange(np.prod(card3), dtype=np.float))
+        tup1 = ('x1', 'x2')
+        tup2 = ('x2', 'x3')
+        tup3 = ('x3', 'x4')
```

Comment:
@abhi95 If you could replace this tuple by something `('x3', (1, 'x4') )`, since we don't restrict variables to have similar types. Try even the cases like where a variable is a tuple and other one a frozenset.